src/jit/lsra.cpp

```diff
@@ -7279,23 +7280,39 @@ void LinearScan::allocateRegisters()
             // then find a register to spill
             if (assignedRegister == REG_NA)
             {
-#ifdef FEATURE_SIMD
+#if FEATURE_PARTIAL_SIMD_CALLEE_SAVE
                 if (refType == RefTypeUpperVectorSaveDef)
                 {
                     // TODO-CQ: Determine whether copying to two integer callee-save registers would be profitable.
-                    currentRefPosition->registerAssignment = (allRegs(TYP_FLOAT) & RBM_FLT_CALLEE_TRASH);
-                    assignedRegister                       = tryAllocateFreeReg(currentInterval, currentRefPosition);
+
+                    // SaveDef position occurs after the Use of args and at the same location as Kill/Def
+                    // positions of a call node.  But SaveDef position cannot use any of the arg regs as
+                    // they are needed for call node.
+                    currentRefPosition->registerAssignment =
+                        (allRegs(TYP_FLOAT) & RBM_FLT_CALLEE_TRASH & ~RBM_FLTARG_REGS);
+                    assignedRegister = tryAllocateFreeReg(currentInterval, currentRefPosition);
+
                     // There MUST be caller-save registers available, because they have all just been killed.
```

Comment:
At first I was concerned that the `LsraLimitSamllFPSet` would not include the registers you need, but then I realized that these assignments are affected. It might be good to add a note to that effect in the comments here.